src/Microsoft.AspNetCore.WebUtilities/FileBufferingReadStream.cs

```diff
@@ -148,7 +160,11 @@ public override bool CanWrite
 
         public override long Length
         {
-            get { return _buffer.Length; }
+            get
+            {
+                var useInnerLength = _innerLength.HasValue && _innerLength > 0 && !_completelyBuffered && _buffer.Position == 0;
```

Comment:
If _innerLength.HasValue, why not always use it?

Reply:
Doing so will break existing tests.

Reply:
Which? Maybe they should be updated instead, this is a behaviour change.

Reply:
See https://travis-ci.org/aspnet/HttpAbstractions/jobs/420667950